src/tools/qt-gui/src/treeviewmodel.cpp

```diff
@@ -489,43 +498,26 @@ void TreeViewModel::populateModel ()
 	populateModel (config);
 }
 
+void TreeViewModel::addRootNode (ElektraNamespace ns)
+{
+	kdb::Key k;
+	k.setNamespace (ns);
+
+	auto name = k.getName ();
+	name = name.substr (0, name.length () - 1);
+
+	m_model << ConfigNodePtr (new ConfigNode (QString::fromStdString (std::string{ static_cast<char> (ns) }),
+						  QString::fromStdString (name), nullptr, this, true));
+}
+
 void TreeViewModel::populateModel (KeySet const & keySet)
 {
 	m_model.clear ();
 
-	using namespace ckdb; // for namespaces
-	for (int i = KEY_NS_FIRST; i <= KEY_NS_LAST; ++i)
-	{
-		elektraNamespace ns = static_cast<elektraNamespace> (i);
-		ConfigNodePtr toAdd;
-		switch (ns)
-		{
-		case KEY_NS_SPEC:
-			toAdd = ConfigNodePtr (new ConfigNode ("spec", "spec", nullptr, this));
-			break;
-		case KEY_NS_PROC:
-			// TODO: add generic commandline parsing
-			break;
-		case KEY_NS_DIR:
-			toAdd = ConfigNodePtr (new ConfigNode ("dir", "dir", nullptr, this));
-			break;
-		case KEY_NS_USER:
-			toAdd = ConfigNodePtr (new ConfigNode ("user", "user", nullptr, this));
-			break;
-		case KEY_NS_SYSTEM:
-			toAdd = ConfigNodePtr (new ConfigNode ("system", "system", nullptr, this));
-			break;
-		case KEY_NS_NONE:
-			break;
-		case KEY_NS_META:
-			break;
-		case KEY_NS_CASCADING:
-			break;
-		case KEY_NS_DEFAULT:
-			break;
-		}
-		if (toAdd) m_model << toAdd;
-	}
+	addRootNode (ElektraNamespace::SPEC);
+	addRootNode (ElektraNamespace::DIR);
+	addRootNode (ElektraNamespace::USER);
+	addRootNode (ElektraNamespace::SYSTEM);
```

Comment:
Why not iterate here over 'static const ElektraNamespace ELEKTRA_NAMESPACES[]'?

Reply:
Because we do not show all namespaces. I find it more readable to list all the namespaces that are shown instead of excluding those that aren't.

Reply:
Can't the namespaces that are not shown be simply removed from the array and the enum then?

Reply:
The general `ELEKTRA_NAMESPACES` array is part of the C++ binding. It has to contain **all** namespaces, not just those that are used in some arbitrary part (the Qt GUI) of Elektra. We could create a separate local array, but I don't see why that would be better.